bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/Event.java

```diff
@@ -357,13 +359,26 @@ public void setBounds (Rectangle rect) {
 	this.height = rect.height;
 }
 
+private int getDeviceZoom() {
+	if (widget != null) {
+		return widget.getZoom();
+	} else if (item != null) {
+		return item.getZoom();
+	} else if (gc != null && gc.getGCData() != null) {
+		return DPIUtil.getZoomForAutoscaleProperty(gc.getGCData().nativeDeviceZoom);
+	} else {
+		return DPIUtil.getDeviceZoom();
```

Comment:
Do you know if/when this case can happen? Using `DPIUtil.getDeviceZoom()` should be something we only do for reasons of compatibility but not because it's actually needed, shouldn't it? I would expect the complete Windows SWT implementation to not use `DPIUtil.getDeviceZoom()` after HiDPI support has been completed at all.

Reply:
Reverted changes for Event, since it is in common. @akoch-yatta has a separate PR for this.